src/lib.rs

```diff
@@ -206,3 +206,15 @@ pub use unit::{DebugTypes, DebugTypesOffset, DebugTypeSignature, TypeUnitHeaders
                TypeUnitHeader};
 pub use unit::{EntriesCursor, EntriesTree, EntriesTreeIter, DebuggingInformationEntry};
 pub use unit::{AttrsIter, Attribute, AttributeValue};
+
+/// A convenience trait for loading DWARF sections from object files.  To be
+/// used like:
+///
+/// fn load_stuff<'a, S: NamedDwarfSection<'a>>() -> S {
+///   let data = thing_that_loads_object_sections<'a>::get(S::name());
+///   S::from(data)
+/// }
+pub trait NamedDwarfSection<'input> : From<&'input [u8]> {
```

Comment:
Does it matter if trait names are generic? If not, I prefer simply `Section` since I think all gimli sections will have names and be for DWARF.